src/coreclr/jit/compiler.h

```diff
@@ -8321,6 +8321,11 @@ XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
     SIMDLevel getSIMDSupportLevel()
     {
 #if defined(TARGET_XARCH)
+        if (compOpportunisticallyDependsOn(InstructionSet_AVX512F))
+        {
+            return SIMD_AVX512F_Supported;
+        }
```

Comment:
I think this should be `compOpportunisticallyDependsOn(InstructionSet_Vector512)`. That covers checking that `F`, `BW`, and `DQ` are all supported and any future changes or tweaks to the required ISAs.

Reply:
Also since this is done as part of `getSIMDVectorRegisterByteLength()`, we're currently "paying" the cost of the check in crossgen and other scenarios.

It might be better to nest this under the AVX2 check instead:
```cpp
if (compOpportunistcallyDependsOn(InstructionSet_Vector256)
{
    if (compOpportunistcallyDependsOn(InstructionSet_Vector512))
    {
        return SIMD_Vector512_Supported;
    }

    // This could be changed to `SIMD_Vector256_Supported`, but since we want to remove the `SIMD_*_Supported`
    // in a follow up PR, it's probably not worth the extra churn. If you were to change it, I'd just go ahead and remove
    // `getSIMDSupportLevel` and do the required `compOpportunistic` checks directly in `getSIMDVectorType` instead
    return SIMD_AVX2_Supported;
}
```

This will make it:
| ISA | # Checks Before | # Checks Now | # Checks With Proposed Change |
| --- | --- | --- | --- |
| AVX512F | N/A | 1 | 2 |
| AVX2 | 1 | 2 | 2 |
| SSE4 | 2 | 3 | 2 |
| SSE2 | 2 | 3 | 2 |

Reply:
Switched to `SIMD_Vector512_Supported` Will leave the avx2 changes to later. Want to keep additional churn low if possible